lib/userAgent.js

```diff
@@ -84,15 +86,22 @@ function getDeviceType (data) {
   }
 }
 
+function getFormFactor (data) {
+  if (data.device && data.device.brand !== 'Generic') {
```

Comment:
Ah yeah, it's covered by the remote tests but that's not obvious because they just give a user agent string and check the result. I'll add local tests to cover it too.